src/main/java/com/minecolonies/colony/Colony.java

```diff
@@ -423,6 +450,22 @@ public void onWorldTick(TickEvent.WorldTickEvent event)
     }
 
     /**
+     * Updates all subscribers of fields etc.
+     */
+    private void markFieldsDirty()
+    {
+        ColonyManager.markDirty();
+
+        for (AbstractBuilding building : buildings.values())
```

Comment:
The mark dirty method should not be the one sending the packets. Thats should happen once per tick if dirty